src/code/padmgr.c

```diff
@@ -1,58 +1,788 @@
+    f32 angle;
+    s8 curX = input->cur.stick_x;
+    s8 curY = input->cur.stick_y;
+
+    if (CHECK_BTN_ANY(input->press.button, BTN_RESET) || input->press.stick_x == 0) {
+        goto dummy;
+    dummy:;
+
+        input->press.stick_x = 61;
+        input->press.errno = -61;
+        input->press.stick_y = 63;
+        input->rel.errno = -63;
+    }
+    pressX = input->press.stick_x;
+    pressX2 = (s8)input->press.errno;
+    pressY = input->press.stick_y;
+    pressY2 = (s8)input->rel.errno;
+
+    if (CHECK_BTN_ANY(input->cur.button, BTN_RESET)) {
+        minus = curX - 7;
+        plus = curX + 7;
+
+        if (minus > 0) {
+            if (pressX < minus - 3) {
+                pressX = minus - 3;
+                input->press.stick_x = minus - 3;
+            }
+        } else if (plus < 0) {
+            if (pressX2 > plus + 3) {
+                pressX2 = plus + 3;
+                input->press.errno = plus + 3;
+            }
+        }
+
+        minus = curY - 7;
+        plus = curY + 7;
+
+        if (minus > 0) {
+            if (pressY < minus - 3) {
+                pressY = minus - 3;
+                input->press.stick_y = minus - 3;
+            }
+        } else if (plus < 0) {
+            if (pressY2 > plus + 3) {
+                pressY2 = plus + 3;
+                input->rel.errno = plus + 3;
+            }
+        }
+    }
+
+    minus = curX - 7;
+    plus = curX + 7;
+
+    if (minus > 0) {
+        newX = (pressX < minus) ? 1.0f : (f32)minus / pressX;
+    } else if (plus < 0) {
+        newX = (pressX2 > plus) ? -1.0f : -(f32)plus / pressX2;
+    } else {
+        newX = 0.0f;
+    }
+
+    minus = curY - 7;
+    plus = curY + 7;
+
+    if (minus > 0) {
+        newY = (pressY < minus) ? 1.0f : (f32)minus / pressY;
+    } else if (plus < 0) {
+        newY = (pressY2 > plus) ? -1.0f : -(f32)plus / pressY2;
+    } else {
+        newY = 0.0f;
+    }
+
+    magnitude = sqrtf(SQ(newX) + SQ(newY));
+    if (magnitude > 1.0f) {
+        magnitude = 1.0f;
+    }
+    angle = Math_Atan2F_XY(curY, -curX);
+    newX = -sinf(angle) * magnitude;
+    newY = cosf(angle) * magnitude;
+dummy2:;
+    input->rel.stick_x = newX * 60.5f;
+    input->rel.stick_y = newY * 60.5f;
+}
+
+/**
+ * Updates `sPadMgrInstance->inputs` based on the error response of each controller
+ */
+void PadMgr_UpdateInputs(void) {
+    s32 diff;
+    Input* input = &sPadMgrInstance->inputs[0];
+    s32 i;
+    OSContPad* pad = &sPadMgrInstance->pads[0];
+
+    for (i = 0; i < sPadMgrInstance->nControllers; i++, input++, pad++) {
+        s32 isStandardController;
+
+        input->prev = input->cur;
+        isStandardController = sPadMgrInstance->ctrlrType[i] == PADMGR_CONT_NORMAL;
+        if (isStandardController) {
+            switch (pad->errno) {
+                case 0:
+                    // No error, copy inputs
+                    input->cur = *pad;
+                    break;
+                case (CHNL_ERR_OVERRUN >> 4):
```

Comment:
Could you add a new line between cases?
```suggestion
                case 0:
                    // No error, copy inputs
                    input->cur = *pad;
                    break;

                case (CHNL_ERR_OVERRUN >> 4):
```